test/test_iterator.cpp

```diff
@@ -263,6 +263,11 @@ namespace sparrow
 
             const_iterator iter{&(a[0])};
             CHECK_EQ(*iter, a[0]);
+
+            iterator iter2{&(a[0])};
+            CHECK_EQ(*iter2, a[0]);
+            *iter2 = 3;
+            CHECK_EQ(a[0], 3);
```

Comment:
This change looks unrelated to the scope of this PR, but it being part of it is OK to me.

Reply:
You right, this tests was missing